polyply/tests/test_gen_seq.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2020 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import math
+import networkx as nx
+from networkx.readwrite import json_graph
+import pytest
+import argparse
+import json
+from pathlib import Path
+import textwrap
+from collections import Counter
+from polyply import gen_seq, TEST_DATA
+from polyply.src.gen_seq import (_add_edges, _branched_graph,
+                                MacroString, generate_seq_graph,
+                               _find_terminal_nodes, _apply_teminii_modifications)
+
+def test_add_edge():
+    graph = nx.Graph()
+    graph.add_nodes_from(list(range(0, 6)))
+    graph.add_edges_from([(0, 1), (1, 2),
+                          (3, 4), (4, 5)])
+    nx.set_node_attributes(graph, {0: 1, 1: 1, 2: 1, 3: 2, 4: 2, 5: 2}, "seqid")
+    edges = "1-2"
+    _add_edges(graph, edges, 1, 2)
+    assert graph.has_edge(1, 5)
+
+@pytest.mark.parametrize("branching_f, n_levels, ref_edges",(
+                        (2, 3, [(0, 1), (0, 2), (1, 3), (1, 4), (2, 5), (2, 6)]),
+                        (1, 4, [(0, 1), (1, 2), (2, 3)])
+                        ))
+def test_branched_graph(branching_f, n_levels, ref_edges):
+    graph = _branched_graph("test", branching_f, n_levels)
+    resnames = {idx: "test" for idx in graph.nodes}
+    assert len(graph.edges) == len(ref_edges)
+    for edge in ref_edges:
+        graph.has_edge(edge[0], edge[1])
+    assert nx.get_node_attributes(graph, "resname") == resnames
+
+
+
+@pytest.mark.parametrize("macro_str, macro_type, ref_edges",(
+                        ("A:3:1:PEO-1.", "linear", [(0, 1), (1, 2)]),
+                        ("A:3:2:PPI-1.", "branched", [(0, 1), (0, 2), (1, 3), (1, 4), (2, 5), (2, 6)]),
```

Comment:
Why the trailing `.` in the strings?

Reply:
because they should be floats not ints. This makes it more clear in the test.

Reply:
Does it really matter? And I would prefer `1.0` over `1.`.